examples/seismic/model.py

```diff
@@ -397,3 +397,105 @@ def critical_dt(self):
         # for further details:
         dt = .85*np.min(self.spacing) / (np.sqrt(self.grid.dim)*self.maxvp)
         return self.dtype("%.3e" % dt)
+
+class ModelViscoacoustic(GenericModel):
+    """
+    The physical model used in seismic inversion processes.
+
+    Parameters
+    ----------
+    origin : tuple of floats
+        Origin of the model in m as a tuple in (x,y,z) order.
+    spacing : tuple of floats
+        Grid size in m as a Tuple in (x,y,z) order.
+    shape : tuple of int
+        Number of grid points size in (x,y,z) order.
+    space_order : int
+        Order of the spatial stencil discretisation.
+    vp : array_like or float
+        Velocity in km/s.
+    rho : float or array, optional
+        Density in kg/cm^3 (rho=1 for water).
+    qp : float or array, optional
+        P-wave quality factor (dimensionless).
+    f0: float or Scalar, optional
+        The dominant frequency
+    nbl : int, optional
+        The number of absorbin layers for boundary damping.
+    dtype : np.float32 or np.float64
+        Defaults to 32.
+
+    The `ModelViscoacoustic` provides two symbolic data objects for the
+    creation of seismic wave propagation operators:
+
+    m : array_like or float
+        The square slowness of the wave.
+    damp : Function
+        The damping field for absorbing boundary condition.
+    """
+    def __init__(self, origin, spacing, shape, space_order, vp, qp, rho, nbl=20, 
+                 subdomains=(), dtype=np.float32):
+        super(ModelViscoacoustic, self).__init__(origin, spacing, shape, space_order,
+                                                nbl=nbl, subdomains=subdomains, 
+                                                dtype=dtype, damp_mask=True)
+
+        self._vp = self._gen_phys_param(vp, 'vp', space_order)
+
+        self.qp = self._gen_phys_param(qp, 'qp', space_order, is_param=True)
+
+        self.rho = self._gen_phys_param(rho, 'rho', space_order, is_param=True)
+
+        self.irho = self._gen_phys_param(1./rho, 'irho', space_order, is_param=True)
```

Comment:
why both? Need one and secong one as symbolic `1/self.rho`  or the opposite.

Reply:
Okay. It really isn't necessary. We will fix it.